modules/apps/frontend-icons/frontend-icons-web/src/main/java/com/liferay/frontend/icons/web/internal/repository/FrontendIconsResourcePackRepository.java

```diff
@@ -0,0 +1,163 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.icons.web.internal.repository;
+
+import com.liferay.document.library.kernel.model.DLFileEntry;
+import com.liferay.document.library.kernel.service.DLFileEntryLocalService;
+import com.liferay.frontend.icons.web.internal.model.FrontendIconsResource;
+import com.liferay.frontend.icons.web.internal.model.FrontendIconsResourcePack;
+import com.liferay.frontend.icons.web.internal.util.SVGUtil;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.model.Repository;
+import com.liferay.portal.kernel.portletfilerepository.PortletFileRepository;
+import com.liferay.portal.kernel.repository.model.FileEntry;
+import com.liferay.portal.kernel.repository.model.Folder;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.util.ContentTypes;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Víctor Galán
+ */
+@Component(service = FrontendIconsResourcePackRepository.class)
+public class FrontendIconsResourcePackRepository {
+
+	public void addFrontendIconsResourcePack(
+			long companyId, FrontendIconsResourcePack frontendIconsResourcePack)
+		throws PortalException {
+
+		Company company = _companyLocalService.getCompany(companyId);
+
+		Folder companyIconsFolder = _getFolder(company);
+
+		String svgSpritemap = SVGUtil.getSVGSpritemap(
+			frontendIconsResourcePack);
+
+		DLFileEntry dlFileEntry = _dlFileEntryLocalService.fetchFileEntry(
+			company.getGroupId(), companyIconsFolder.getFolderId(),
+			frontendIconsResourcePack.getName());
+
+		FileEntry fileEntry = _portletFileRepository.fetchPortletFileEntry(
+			company.getGroupId(), companyIconsFolder.getFolderId(),
+			frontendIconsResourcePack.getName());
+
+		if (dlFileEntry != null) {
+			_portletFileRepository.deletePortletFileEntry(
+				fileEntry.getFileEntryId());
+		}
+
+		_portletFileRepository.addPortletFileEntry(
+			company.getGroupId(), _userLocalService.getDefaultUserId(companyId),
+			null, 0, _REPOSITORY_NAME, companyIconsFolder.getFolderId(),
+			svgSpritemap.getBytes(), frontendIconsResourcePack.getName(),
+			ContentTypes.IMAGE_SVG_XML, false);
+	}
+
+	public void deleteIconResourcePack(long companyId, String iconPackName)
+		throws PortalException {
+
+		Company company = _companyLocalService.getCompany(companyId);
+
+		Folder companyIconsFolder = _getFolder(company);
```

Comment:
How about calling the method `_getCompanyIconsFolder(company)` (as per the variable)?